chains/evm/calls/transactor/itx/itx.go

```
@@ -0,0 +1,163 @@
+func (itx *ITXTransactor) Transact(to *common.Address, data []byte, opts transactor.TransactOptions) (*common.Hash, error) {
+	err := transactor.MergeTransactionOptions(&opts, &DefaultTransactionOptions)
+	if err != nil {
+		return nil, err
+	}
+	opts.ChainID = itx.forwarder.ChainId()
+
+	defer itx.forwarder.UnlockNonce()
+	itx.forwarder.LockNonce()
+
+	nonce, err := itx.forwarder.UnsafeNonce()
+	if err != nil {
+		return nil, err
+	}
+	opts.Nonce = nonce
+
+	forwarderData, err := itx.forwarder.ForwarderData(to, data, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	// increase gas limit because of forwarder overhead
+	opts.GasLimit = opts.GasLimit * 11 / 10
+	signedTx, err := itx.signRelayTx(&RelayTx{
+		to:   itx.forwarder.ForwarderAddress(),
+		data: forwarderData,
+		opts: opts,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	h, err := itx.sendTransaction(context.Background(), signedTx)
+	if err != nil {
+		return nil, err
+	}
+
+	itx.forwarder.UnsafeIncreaseNonce()
+	return &h, nil
+}
+
+func (itx *ITXTransactor) signRelayTx(tx *RelayTx) (*SignedRelayTx, error) {
+	uint256Type, _ := abi.NewType("uint256", "uint256", nil)
+	addressType, _ := abi.NewType("address", "address", nil)
+	bytesType, _ := abi.NewType("bytes", "bytes", nil)
+	stringType, _ := abi.NewType("string", "string", nil)
+	arguments := abi.Arguments{
+		{Type: addressType},
+		{Type: bytesType},
+		{Type: uint256Type},
+		{Type: uint256Type},
+		{Type: stringType},
+	}
+	packed, err := arguments.Pack(
+		tx.to,
+		tx.data,
+		big.NewInt(int64(tx.opts.GasLimit)),
+		tx.opts.ChainID,
+		tx.opts.Priority,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	txID := crypto.Keccak256Hash(packed)
+	msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(txID), string(txID.Bytes()))
+	hash := crypto.Keccak256Hash([]byte(msg))
+	sig, err := crypto.Sign(hash.Bytes(), itx.kp.PrivateKey())
+	if err != nil {
+		return nil, nil
```

You should probably return the error?